subprojects/core/src/main/java/org/gradle/internal/service/scopes/GradleUserHomeScopeServices.java

```diff
@@ -107,6 +110,8 @@
  * Defines the shared services scoped to a particular Gradle user home directory. These services are reused across multiple builds and operations.
  */
 public class GradleUserHomeScopeServices {
+    public static final String ENABLE_VFS_SYSTEM_PROPERTY_NAME = "org.gradle.experimental.enable.vfs";
```

Comment:
Can we do this via a `FeaturePreviews.Feature`?

Reply:
We can't enable which virtual file system implementation to use via a `FeaturePreview`, since `GradleUserHomeScopeServices` are started before we parse the settings file.

Later, when we only switch fine grained validation on and off, we can use a feature preview, since that is only relevant for task execution which happens below build session scope.